src/main/java/org/eclipse/microprofile/starter/log/DynamoDBLogger.java

```diff
@@ -0,0 +1,318 @@
+            Log.log(Level.SEVERE, "HttpURLConnection failed while reading error stream.", e);
+            return Status.NOK;
+        }
+
+        if (Log.isLoggable(Level.FINE)) {
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(myURLConnection.getInputStream()))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    Log.log(Level.FINE, line);
+                }
+            } catch (IOException e) {
+                Log.log(Level.SEVERE, "HttpURLConnection failed while reading input stream.", e);
+                return Status.NOK;
+            }
+        }
+
+        return Status.OK;
+    }
+
+    private static void validateEngineData(final EngineData engineData) {
+        if (engineData.getMpVersion() == null || engineData.getMpVersion().length() < 1) {
+            throw new IllegalArgumentException("mpVersion must not be null/empty.");
+        }
+        if (engineData.getSupportedServer() == null || engineData.getSupportedServer().length() < 1) {
+            throw new IllegalArgumentException("supportedServer must not be null/empty.");
+        }
+        if (engineData.getBeansxmlMode() == null || engineData.getBeansxmlMode().length() < 1) {
+            throw new IllegalArgumentException("beansxmlMode must not be null/empty.");
+        }
+        if (engineData.getSelectedSpecs() == null) {
+            throw new IllegalArgumentException("selectedSpecs must not be null");
+        }
+        if (engineData.getMavenData().getArtifactId() == null || engineData.getMavenData().getArtifactId().length() < 1) {
+            throw new IllegalArgumentException("artifacId must not be null/empty.");
+        }
+        if (engineData.getMavenData().getGroupId() == null || engineData.getMavenData().getGroupId().length() < 1) {
+            throw new IllegalArgumentException("groupId must not be null/empty.");
+        }
+    }
+
+    /**
+     * Expensive blocking method that creates signed request and does HTTP POST to AWS DynamoDB API
+     *
+     * @param engineData Logged as is except for groupIdArtifacId which is merely used to enhance log message hash.
+     * @return Either Status.OK or Status.NOK if the AWS DynamoDB access is configured.
+     * The caller is not expected to be able to do anything about the error anyway.
+     */
+    public Status log(final EngineData engineData) {
+        if (accessKey == null || (accessKey.length()) < 10) {
+            Log.log(Level.FINE, "System property AWS_ACCESS_KEY_ID not defined. Log call dropped.");
+            return Status.NOT_CONFIGURED;
+        }
+        if (secretKey == null || (secretKey.length()) < 32) {
+            Log.log(Level.FINE, "System property AWS_SECRET_ACCESS_KEY not defined. Log call dropped.");
+            return Status.NOT_CONFIGURED;
+        }
+
+        validateEngineData(engineData);
+
+        final Calendar calendar = Calendar.getInstance();
```

Comment:
Use new Date() since we are not using anything specific from Calendar except Calendar.getTime() which return the Date Object

Reply:
@rdebusscher Ack.